modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/QuerySpecTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.rankeval;
+
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ParseFieldRegistry;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.query.QueryParseContext;
+import org.elasticsearch.indices.query.IndicesQueriesRegistry;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.search.aggregations.AggregatorParsers;
+import org.elasticsearch.search.suggest.Suggesters;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.util.List;
+
+public class QuerySpecTests extends ESTestCase {
+
+    private static IndicesQueriesRegistry queriesRegistry;
+    private static SearchModule searchModule;
+    private static Suggesters suggesters;
+    private static AggregatorParsers aggsParsers;
+
+    /**
+    * setup for the whole base test class
+    */
+    @BeforeClass
+    public static void init() throws IOException {
+        aggsParsers = new AggregatorParsers(new ParseFieldRegistry<>("aggregation"), new ParseFieldRegistry<>("aggregation_pipes"));
+        searchModule = new SearchModule(Settings.EMPTY, new NamedWriteableRegistry(), false);
+        queriesRegistry = searchModule.getQueryParserRegistry();
+        suggesters = searchModule.getSuggesters();
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        queriesRegistry = null;
+        searchModule = null;
+        suggesters = null;
+        aggsParsers = null;
+    }
+
+    public void testParseFromXContent() throws IOException {
```

Comment:
Should we have some sort of round trip parsing testing like we added during the query refactoring? Is that even possible?